internal/pkg/service/stream/api/mapper/page.go

```diff
@@ -0,0 +1,99 @@
+package mapper
+
+import (
+	"context"
+	"strings"
+
+	etcd "go.etcd.io/etcd/client/v3"
+
+	streamDesign "github.com/keboola/keboola-as-code/api/stream"
+	svcerrors "github.com/keboola/keboola-as-code/internal/pkg/service/common/errors"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/iterator"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/common/etcdop/op"
+	"github.com/keboola/keboola-as-code/internal/pkg/service/stream/api/gen/stream"
+	"github.com/keboola/keboola-as-code/internal/pkg/utils/errors"
+)
+
+// loadPage using etcd iterator and map records to API results.
+// E is type of the entity from the database.
+// R is type of the mapped response.
+func loadPage[E, R any](
+	ctx context.Context,
+	sinceID string,
+	limit int,
+	sort etcd.SortOrder,
+	list func(...iterator.Option) iterator.DefinitionT[E],
+	mapper func(E) R,
+) (out []R, page *stream.PaginatedResponse, err error) {
+	// Check limits
+	if sort != etcd.SortAscend && sort != etcd.SortDescend {
+		return nil, nil, errors.New(`sort must be etcd.SortAscend os etcd.SortDescend`)
```

Comment:
typo  'os' ->  'or'

Reply:
:+1: Fixed in https://github.com/keboola/keboola-as-code/pull/1641 / https://github.com/keboola/keboola-as-code/pull/1641/commits/0dec7af1305bcb4e7c91d253d3f28065c4434889